Self Service App / Mifos Mobile Crashes on Samsung with  Android version 4.4.2

**Summary:**
@therajanmaurya @dilpreet96 
When I tried to run the app from the android studio building apk from the latest source code development branch builds well and installs on Samsung but launching it crashes.
with a message "Unfortunately Mifos Mobile has stopped"
I tried disabling instant run but it does not work.

**Log Cat Errors:**
10-01 16:55:56.275 26645-26645/? E/dalvikvm: >>>>> Normal User
    >>>>> org.mifos.mobilebanking [ userId:0 | appId:10214 ]
10-01 16:55:56.355 26645-26645/org.mifos.mobilebanking E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mifos.mobilebanking, PID: 26645
    java.lang.RuntimeException: Unable to get provider com.google.firebase.provider.FirebaseInitProvider: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.provider.FirebaseInitProvider" on path: DexPathList[[zip file "/data/app/org.mifos.mobilebanking-4.apk"],nativeLibraryDirectories=[/data/app-lib/org.mifos.mobilebanking-4, /vendor/lib, /system/lib]]
        at android.app.ActivityThread.installProvider(ActivityThread.java:5218)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:4810)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4693)
        at android.app.ActivityThread.access$1600(ActivityThread.java:173)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1363)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:136)
        at android.app.ActivityThread.main(ActivityThread.java:5579)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:515)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1268)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1084)
        at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.provider.FirebaseInitProvider" on path: DexPathList[[zip file "/data/app/org.mifos.mobilebanking-4.apk"],nativeLibraryDirectories=[/data/app-lib/org.mifos.mobilebanking-4, /vendor/lib, /system/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:67)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
        at android.app.ActivityThread.installProvider(ActivityThread.java:5203)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:4810) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4693) 
        at android.app.ActivityThread.access$1600(ActivityThread.java:173) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1363) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:136) 
        at android.app.ActivityThread.main(ActivityThread.java:5579) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:515) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1268) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1084) 
        at dalvik.system.NativeStart.main(Native Method) 


**Device and Android version:**

Model device: Samsung Galaxy S5 What Android
Model version: 4.4.2

--- a/app/src/androidTest/java/org/mifos/mobile/ApplicationTest.java
+++ b/app/src/androidTest/java/org/mifos/mobile/ApplicationTest.java
@@ -19,6 +19,6 @@ public class ApplicationTest {
     public void useAppContext() {
         // Context of the app under test.
         Context appContext = InstrumentationRegistry.getTargetContext();
-        assertEquals("org.mifos.mobilebanking", appContext.getPackageName());
+        assertEquals("org.mifos.mobile", appContext.getPackageName());
     }
 }

--- a/app/src/main/java/org/mifos/mobile/MifosSelfServiceApp.java
+++ b/app/src/main/java/org/mifos/mobile/MifosSelfServiceApp.java
@@ -1,6 +1,5 @@
 package org.mifos.mobile;
 
-import android.app.Application;
 import android.content.Context;
 
 import com.crashlytics.android.Crashlytics;
@@ -14,13 +13,15 @@ import org.mifos.mobile.injection.module.ApplicationModule;
 import org.mifos.mobile.utils.LanguageHelper;
 
 import androidx.appcompat.app.AppCompatDelegate;
+import androidx.multidex.MultiDex;
+import androidx.multidex.MultiDexApplication;
 import io.fabric.sdk.android.Fabric;
 
 /**
  * @author ishan
  * @since 08/07/16
  */
-public class MifosSelfServiceApp extends Application {
+public class MifosSelfServiceApp extends MultiDexApplication {
 
     ApplicationComponent applicationComponent;
 
@@ -41,6 +42,7 @@ public class MifosSelfServiceApp extends Application {
     @Override
     public void onCreate() {
         super.onCreate();
+        MultiDex.install(this);
         Fabric.with(this, new Crashlytics());
         instance = this;
         FlowManager.init(new FlowConfig.Builder(this).build());

--- a/app/src/main/res/drawable/ic_wrapped_lock_black_24dp.xml
+++ b/app/src/main/res/drawable/ic_wrapped_lock_black_24dp.xml
@@ -1,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/ic_lock_black_24dp"/>
+</layer-list>

--- a/app/src/main/res/drawable/ic_wrapped_person_black_24dp.xml
+++ b/app/src/main/res/drawable/ic_wrapped_person_black_24dp.xml
@@ -1,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/ic_person_black_24dp"/>
+</layer-list>

--- a/app/src/main/res/layout/activity_login.xml
+++ b/app/src/main/res/layout/activity_login.xml
@@ -40,9 +40,9 @@
 
                 <EditText
                     style="@style/Base.TextAppearance.AppCompat.Medium"
-                    android:drawableLeft="@drawable/ic_person_black_24dp"
+                    android:drawableLeft="@drawable/ic_wrapped_person_black_24dp"
                     android:drawablePadding="@dimen/layout_padding_8dp"
-                    android:drawableStart="@drawable/ic_person_black_24dp"
+                    android:drawableStart="@drawable/ic_wrapped_person_black_24dp"
                     android:hint="@string/username"
                     android:inputType="textEmailAddress"
                     android:layout_height="wrap_content"
@@ -61,9 +61,9 @@
 
                 <EditText
                     style="@style/Base.TextAppearance.AppCompat.Medium"
-                    android:drawableLeft="@drawable/ic_lock_black_24dp"
+                    android:drawableLeft="@drawable/ic_wrapped_lock_black_24dp"
                     android:drawablePadding="@dimen/layout_padding_8dp"
-                    android:drawableStart="@drawable/ic_lock_black_24dp"
+                    android:drawableStart="@drawable/ic_wrapped_lock_black_24dp"
                     android:hint="@string/password"
                     android:inputType="textPassword"
                     android:layout_height="wrap_content"

